bot/cogs/help.py

```diff
@@ -0,0 +1,236 @@
+from inspect import Parameter
+import textwrap
+import typing as t
+
+import discord
+from discord.ext import commands, menus
+import discord.utils
+
+
+class HelpSource(menus.ListPageSource):
+    """The Help menu."""
+
+    def __init__(
+            self,
+            signature: t.Callable,
+            filter_commands: t.Coroutine,
+            prefix: str,
+            author: discord.User,
+            cogs: t.Dict[t.Optional[commands.Cog], t.List[commands.Command]],
+    ) -> None:
+        """Create the menu."""
+        self.get_command_signature = signature
+        self.filter_commands = filter_commands
+        self.prefix = prefix
+        self.menu_author = author
+        sorted_cogs = sorted(
+            cogs,
+            key=lambda cog: cog.qualified_name if cog else "ZZ"
+        )
+        super().__init__(
+            [(cog, cogs[cog]) for cog in sorted_cogs],
+            per_page=1,
+        )
+
+    async def format_page(
+            self,
+            menu: menus.Menu,
+            cog_tuple: t.Tuple[
+                t.Optional[commands.Cog], t.List[commands.Command]
+            ],
+    ) -> discord.Embed:
+        """Format the pages."""
+        cog, command_list = cog_tuple
+        embed = discord.Embed(
+            title=textwrap.dedent(
+                f"""
+                Help for
+                {cog.qualified_name if cog else 'unclassified commands'}
+                """
+            ),
+            description=textwrap.dedent(
+                f"""
+                Help syntax : `<Required argument>`. `[t.Optional argument]`
+                Command prefix: `{self.prefix}`
+                {cog.description if cog else ""}
+                """
+            ),
+            color=0xffff00,
+        )
+        embed.set_author(
+            name=self.menu_author.display_name,
+            icon_url=str(self.menu_author.avatar_url),
+        )
+        for command in await self.filter_commands(command_list):
+            embed.add_field(
+                name=f"{self.prefix}{self.get_command_signature(command)}",
+                value=command.help,
+                inline=False,
+            )
+        embed.set_footer(
+            text=f"Page {menu.current_page+1}/{self.get_max_pages()}"
+        )
+        return embed
+
+
+class Help(commands.HelpCommand):
+    """The Help cog."""
```

Comment:
This class is not a Cog, this docstring is faulty

Reply:
Oh wupsy